meson.build

```diff
@@ -0,0 +1,218 @@
+
+c_warn_flags = '''
+  -Wundef
+  -Wnested-externs
+  -Wwrite-strings
+  -Wpointer-arith
+  -Wmissing-declarations
+  -Wmissing-prototypes
+  -Wstrict-prototypes
+  -Wredundant-decls
+  -Wno-unused-parameter
+  -Wno-missing-field-initializers
+  -Wdeclaration-after-statement
+  -Wformat=2
+  -Wold-style-definition
+  -Wcast-align
+  -Wformat-nonliteral
+  -Wformat-security
+  -Wsign-compare
+  -Wstrict-aliasing
+  -Wshadow
+  -Winline
+  -Wpacked
+  -Wmissing-format-attribute
+  -Wmissing-noreturn
+  -Winit-self
+  -Wmissing-include-dirs
+  -Wunused-but-set-variable
+  -Warray-bounds
+  -Wimplicit-function-declaration
+  -Wreturn-type
+  -Wswitch-enum
+  -Wswitch-default
+  -Wno-error=unused-parameter
+  -Wno-error=missing-field-initializers
+'''.split()
+# To allow building against OpenSSL 3.0, we still need to allow deprecated
+# declarations
+c_warn_flags += '-Wno-error=deprecated-declarations'
+c_flags += cc.get_supported_arguments(c_warn_flags)
+c_flags += ['-fdata-sections', '-ffunction-sections', '-fno-strict-aliasing']
+c_flags += ['-Wl,--gc-sections', '-Wl,-Map,$@.map']
+add_project_arguments(c_flags, language : 'c')
+# set logging domain to 'rauc'
+add_project_arguments('-DG_LOG_DOMAIN="rauc"', language : 'c')
+# enable GNU extensions (required!)
+add_project_arguments('-D_GNU_SOURCE', language : 'c')
+
+incdir = include_directories('include')
+
+sources_rauc = files([
+  'src/bootchooser.c',
+  'src/bundle.c',
+  'src/checksum.c',
+  'src/config_file.c',
+  'src/context.c',
+  'src/crypt.c',
+  'src/dm.c',
+  'src/emmc.c',
+  'src/hash_index.c',
+  'src/install.c',
+  'src/manifest.c',
+  'src/mark.c',
+  'src/mbr.c',
+  'src/mount.c',
+  'src/service.c',
+  'src/signature.c',
+  'src/stats.c',
+  'src/slot.c',
+  'src/update_handler.c',
+  'src/utils.c',
+  'src/verity_hash.c',
+])
+
+conf.set10('ENABLE_NETWORK', get_option('network'))
+if get_option('network')
+  sources_rauc += files('src/network.c')
+endif
+
+if (get_option('streaming') and not get_option('network'))
+  error('streaming support requested but network support not enabled')
+endif
+conf.set10('ENABLE_STREAMING', get_option('streaming'))
+if get_option('streaming')
+  if not cc.has_header('linux/nbd-netlink.h')
+    error('streaming support requested but linux/nbd-netlink.h not found')
+  endif
+  sources_rauc += files('src/nbd.c')
+endif
+
+conf.set10('ENABLE_EMMC_BOOT_SUPPORT', cc.has_header('linux/mmc/ioctl.h'))
+
+conf.set10('ENABLE_GPT', fdiskdep.found())
+if fdiskdep.found()
+  sources_rauc += files('src/gpt.c')
+endif
+
+gnome = import('gnome')
+dbus_ifaces = files('src/de.pengutronix.rauc.Installer.xml')
+dbus_sources = gnome.gdbus_codegen(
+  'rauc-installer-generated',
+  sources : dbus_ifaces,
+  interface_prefix : 'de.pengutronix.rauc.',
+  namespace : 'R',
+  install_header : true,
+)
+
+config_h = configure_file(
+  output : 'config.h',
+  configuration : conf
+)
+add_project_arguments('-include' + meson.current_build_dir() / 'config.h', language : 'c')
```

Comment:
Can't we use the file object here?

Reply:
I attempted to do so but this does not support obtaining the (absolute) file name as a string (which would be required here)